docs/design/kata-nydus-design.md

```diff
@@ -0,0 +1,89 @@
+# Background
+
+[Research](https://www.usenix.org/conference/fast16/technical-sessions/presentation/harter) shows that time to take for pull operation accounts for 76% of container startup time but only 6.4% of that data is read. So if we can get data on demand (lazy load), it will speed up the container start. [Nydus](https://github.com/dragonflyoss/image-service) is a project which build image with new format and can get data on demand when container start. Nydus has supported runc containers.
+
+The following benchmarking result shows the performance improvement compared with the OCI image for the container cold startup elapsed time on containerd. As the OCI image size increases, the container startup time of using Nydus image remains very short.
+
+![perf](arch-images/perf.jpeg)
+
+## Proposal - Bring lazyload ability to Kata Containers
+
+Nydusd is a fuse/virtiofs daemon which is provided by nydus project and it supports Passthroughfs and [RAFS](https://github.com/dragonflyoss/image-service/blob/master/docs/nydus-design.md) (Registry Acceleration File System) natively, so in Kata Containers, we can use nydusd in place of virtiofsd and mount nydus image to guest in the meanwhile.
+
+The process of create/start Kata Containers with virtiofsd,
+
+1. when create sandbox, the Kata Containers shim will launch virtiofsd before VM start, so VM can share dir with host.
+2. when create container, the Kata Containers shim will mount rootfs to /path/to/bundle/rootfs and bind mount rootfs to `sharedir`, so it can be seen in the guest.
+
+The process of create/start Kata Containers with nydusd,
+
+![kata-nydus](arch-images/kata-nydus.png)
+
+1. when create sandbox, the Kata Containers Containerd v2 shim start `nydusd` daemon and mount `passthroughfs` filesystem before the VM starts.
+
+```bash
+# start nydusd
+$ sudo /usr/local/bin/nydusd --log-level info  --sock /run/vc/vm/5502d6f9a404ccca94320d7d47f9225e25e05d193d521d713565c01811e480d5/vhost-user-fs.sock --apisock  /run/vc/vm/5502d6f9a404ccca94320d7d47f9225e25e05d193d521d713565c01811e480d5/api.sock
```

Comment:
This line is too long to read in markdown preview mode, suggest breaking it to multiple lines like `curl` command below.

Reply:
done, use ${sandboxID} instead